### PR TITLE
IOS-2066: Converted SHAutocorrectSuggestionView into Swift

### PR DESCRIPTION
**Pull request by bdrelling**
**Issue Link**
IOS-2066

**Description**
- Converted `SHAutocorrectSuggestionView` from Obj-C to Swift. This is a 1:1 conversion with no intentional stylistic changes other than using a `guard` where it saved us from nesting over and over. That includes leaving methods be and not converting them to computed vars when appropriate. My heart is heavy and my soul hurts.
- Added a ton of inline documentation for `SHAutocorrectSuggestionView.swift` showing the Obj-C code it was translated from. Will keep it around for a while. Added a note saying as much as well.

### CODE REVIEW

**Comment by bdrelling** [comment]
Still not quite finished, one more function to go.

**Comment by bdrelling** [comment]
Still working through an issue with the conversion:

Simulator Screen Shot - iPhone 11 Pro - 2020-05-08 at 15 33 58

Something with the sizing of the font. Looking through the code to see what got missed.

File: Sources/SpotHeroEmailValidator/SHAutocorrectSuggestionView.swift [hunk 1]
```diff
@@ -1,41 +1,455 @@
...
+                                                attributes: [
+                                                    .font: self.titleFont,
+                                                    .paragraphStyle: paragraphTitleStyle,
+                                                    .foregroundColor: UIColor.white,
+                                                ],
+                                                context: nil)
+        
+        // CGRect suggestionSizeRect = [suggestion boundingRectWithSize:CGSizeMake(kMaxWidth - kDismissButtonWidth, MAXFLOAT) options:NSStringDrawingUsesLineFragmentOrigin attributes:@{NSFontAttributeName:self.suggestionFont, NSParagraphStyleAttributeName:paragraphSuggestedStyle, NSForegroundColorAttributeName:[SHAutocorrectSuggestionView defaultSuggestionColor]} context:nil];
+        
+        let suggestionSizeRect = title?.boundingRect(with: CGSize(width: Self.maxWidth - Self.dismissButtonWidth, height: CGFloat.greatestFiniteMagnitude),
```
Comment (joevandeventer) [inline Sources/SpotHeroEmailValidator/SHAutocorrectSuggestionView.swift:139]: Should switch this to `suggestion?.boundingRectWithSize`

**Comment by bdrelling** [comment]
@joevandeventer my hero! thanks

**Pull request merged**